FirstSolution/Part17Task2/MathOperations.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+
+namespace Part17Task2
+{
+    public class MathOperations
+    {
+        public static double MathOp(double x, double y, Operation op)
+        {
+            double result = 0;
+
+            try
+            {
+                switch (op)
+                {
+                    case Operation.Addition:
+                        result = x + y;
+                        break;
+                    case Operation.Subtraction:
+                        result = x - y;
+                        break;
+                    case Operation.Multiplication:
+                        result = x * y;
+                        break;
+                    case Operation.Division:
+                        if (y == 0)
+                        {
+                            throw new DivideByZeroException();
+                        }
+                        result = x / y;
```

Comment:
You are right with DivideByZeroException, but try to remove this check and see how it works without it.

Reply:
We have just discussed this problem